### PR TITLE
Refactor some validation and error handling code

### PR DESCRIPTION
**Pull request by dsgoers**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,13 +3,25 @@
 class ApplicationController < ActionController::Base
   before_action :decode_json_api
 
-  rescue_from ActiveRecord::RecordNotFound do |_exception|
-    render_error(ApiError.new(:id, 'Not found.'), :not_found)
+  rescue_from ActiveRecord::RecordNotFound do |exception|
+    render_api_error(exception.to_s, :not_found)
```
Comment (twinge) [inline app/controllers/application_controller.rb]: Is there a reason for using `exception.to_s` here and `exception.message` in the other cases below? I'm assuming they do the same thing, in which case consistency is nice.

**Comment by dsgoers** [comment]
Wow, that was a fast review

**Comment by dsgoers** [comment]
thanks!!!

**Comment by twinge** [comment]
yw

**Pull request merged**